core/include/scipp/core/except.h

```diff
@@ -104,6 +104,19 @@ template <class A, class B> void equals(const A &a, const B &b) {
     throw scipp::except::MismatchError(a, b);
 }
 
+template <class A, class Dim, class System, class Enable>
+void equals(const A &a, const boost::units::unit<Dim, System, Enable> &unit) {
```

Comment:
Why do we need `boost::units::unit` as an argument instead of units::Unit?

Reply:
`units::Unit` doesn't seem to implicitly convert to `boost::units::unit`. Passing a `units::dimensionless` to a argument of `units::Unit` doesn't compile.

I am not sure how to make the conversion work, but if possible then these new equals functions can be removed as the `units::dimensionless` will be match the other `equals` overload.

Reply:
Yes, it is a problem, we should solve it but not here. But better move this somewhere closer to units, I mean this shouldn't be in the core. If it is related to exceptions somehow I would suggest `scipp/units/include/scipp/units/except.h`, but I can't see if it is the case.

Reply:
That did not work as `boost::units::unit` does not seem to have a `.name()` method unless converted to a `units::Unit`. The `to_string` would have to convert it to a `Unit` before finding the name, which is what the new `equals` do.

Reply:
@SimonHeybrock, we should fix this anyway, remember the case with scalar * unit, there you should explicitly cast to Unit: https://github.com/scipp/scipp/blob/4e6d5dd5e55679879a92b4284a7357b19359c0f9/core/test/variable_test.cpp#L966-L967

Reply:
What I meant is:

```cpp
template <class A, class Dim, class System, class Enable>
MismatchError(const units::Unit &, const boost::units::unit<Dim, System, Enable> &)->MismatchError<units::Unit>; 
```
... does this make the conversion work?

Edit: No, because the deduction does not change the constructor template param: https://github.com/scipp/scipp/blob/d4738d1ffd8a161c96d1df69e97decf4d5127bb6/common/include/scipp/common/except.h#L19-L22

What if we change that to `to_string(T(b)`? Or does that break other bits?

Reply:
`to_string(T(b))` works. But wouldn't that trigger a copy for `Variable`, `Dataset` that could potentially be very costly? What about a `to_string` overload for boost units?

Reply:
You are probably right. Either keep what you have, or add the `to_string` you suggest, we should not let this hold up the PR.